meson.build

```diff
@@ -0,0 +1,123 @@
+project('libtexpdf', 'c', version : '2.4.6', default_options : ['c_std=gnu11'])
+
+pkg = import('pkgconfig')
+
+compiler = meson.get_compiler('c')
+
+conf = configuration_data()
+conf.set('version', meson.project_version())
+
+check_headers = ['dlfcn.h', 'inttypes.h', 'stdint.h', 'stdlib.h', 'string.h', 'sys/stat.h', 'sys/types.h', 'sys/wait.h', 'unistd.h']
+foreach header_name : check_headers
+    if compiler.has_header(header_name)
+        conf.set('HAVE_' + header_name.underscorify().to_upper(), 1)
+    endif
+endforeach
+
+if compiler.has_function('timezone', prefix : '#include <time.h>')
+    conf.set('HAVE_TIMEZONE', 1)
+endif
+
+if compiler.has_function('getenv', prefix : '#include <stdlib.h>')
+    conf.set('HAVE_GETENV', 1)
+endif
+
+if compiler.has_function('mkstemp', prefix : '#include <stdlib.h>')
+    conf.set('HAVE_MKSTEMP', 1)
+endif
+
+if compiler.has_member('struct tm', 'tm_gmtoff', prefix : '#include <time.h>')
+    conf.set('HAVE_TM_GMTOFF', 1)
+endif
+
+configure_file(input: 'config.h.meson', output: 'config.h', configuration: conf)
+
+libm = compiler.find_library('m', required : false)
+libdl = compiler.find_library('dl', required : false)
+
+zlib_dep = dependency('zlib')
+libpng_dep = dependency('libpng')
```

Comment:
This is... interesting, they are mandatory in configure.ac but not in the source code.

Reply:
I'm going to take a *guess* at this. The original code here is ripped out of a different context where it might have been optional, but the configure script has been tweaked specifically for use as a stand alone library by SILE, where support for these is required. Hence what the configure script states as requirements is probably more canonical than what the source might do.

Reply:
> I'm going to take a _guess_ at this. The original code here is ripped out of a different context where it might have been optional, but the configure script has been tweaked specifically for use as a stand alone library by SILE, where support for these is required. Hence what the configure script states as requirements is probably more canonical than what the source might do.

This was my guess too. Probably a later modification to the code can be done to use these library unconditionally since they are required anyway but for this PR I think its better to limit its scope to the introduction of a Meson build.